packages/core/src/lib/Cache.ts

```diff
@@ -4,158 +4,121 @@ import { ICategory } from "./Category.js"
-      current.updateIfNewer(supersedes, entry)
+      current.updateIfNewer(entry)
     }
   }
 }
 
 export interface IReloadableCategory<E, S, C> extends ICategory<E, S, C> {
   reload(streamName: string, requireLeader: boolean, t: TokenAndState<S>): Promise<TokenAndState<S>>
-  supersedes: Supersedes
 }
 
 export interface ICachingStrategy {
-  load<S>(streamName: string): Promise<TokenAndState<S> | undefined>
-  store<S>(
-    supersedes: Supersedes,
-    streamName: string,
-    value: Promise<TokenAndState<S>>,
-  ): Promise<TokenAndState<S>>
-}
-
-class NoCaching implements ICachingStrategy {
-  async load<S>(): Promise<TokenAndState<S> | undefined> {
-    return undefined
-  }
-  store<S>(
-    _supersedes: Supersedes,
-    _streamName: string,
-    value: Promise<TokenAndState<S>>,
-  ): Promise<TokenAndState<S>> {
-    return value
-  }
-}
-
-class SlidingWindow implements ICachingStrategy {
-  constructor(
-    private readonly cache: ICache,
-    private readonly windowInMs: number,
-    private readonly prefix = "",
-  ) {}
-
-  load<S>(streamName: string): Promise<TokenAndState<S> | undefined> {
-    return this.cache.tryGet(this.prefix + streamName)
-  }
-
-  async store<S>(
-    supersedes: Supersedes,
-    streamName: string,
-    value: Promise<TokenAndState<S>>,
-  ): Promise<TokenAndState<S>> {
-    const v = await value
-    const key = this.prefix + streamName
-    const expiration = { relative: this.windowInMs }
-    await this.cache.updateIfNewer(key, expiration, supersedes, CacheEntry.ofTokenAndState(v))
-    return v
-  }
-}
-
-class FixedTimeSpan implements ICachingStrategy {
-  constructor(
-    private readonly cache: ICache,
-    private readonly timeSpanInMs: number,
-    private readonly prefix = "",
-  ) {}
-
-  load<S>(streamName: string): Promise<TokenAndState<S> | undefined> {
-    return this.cache.tryGet(this.prefix + streamName)
-  }
-
-  async store<S>(
-    supersedes: Supersedes,
-    streamName: string,
-    value: Promise<TokenAndState<S>>,
-  ): Promise<TokenAndState<S>> {
-    const v = await value
-    const key = this.prefix + streamName
-    const expiration = { absolute: Date.now() + this.timeSpanInMs }
-    await this.cache.updateIfNewer(key, expiration, supersedes, CacheEntry.ofTokenAndState(v))
-    return v
-  }
+  cache: ICache
+  cacheKey: (streamName: string) => string
 }
 
 export namespace CachingStrategy {
-  export const slidingWindow = (
-    cache: ICache,
-    windowInMs: number,
-    prefix?: string,
-  ): ICachingStrategy => new SlidingWindow(cache, windowInMs, prefix)
-  export const noCache = (): ICachingStrategy => new NoCaching()
-  export const fixedTimeSpan = (
-    cache: ICache,
-    periodInMs: number,
-    prefix?: string,
-  ): ICachingStrategy => new FixedTimeSpan(cache, periodInMs, prefix)
+  export const Cache = (cache: ICache, prefix?: string): ICachingStrategy => ({
+    cache,
+    cacheKey: (streamName: string) => (prefix || "") + streamName,
```

Comment:
`prefix === null ? streamName : <concat>` unless you are completly sure this is mapped more efficiently in all cases, given that no prefix is the 99.99% case